sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php

```diff
@@ -0,0 +1,313 @@
+   * @param object $node
+   *   The node.
+   *
+   * @return string|null
+   *   The body text value if any.
+   */
+  public function getBody($node) {
+    return $this->getTextFieldValue($node, 'body', NULL, FALSE);
+  }
+
+  /**
+   * Get image url.
+   *
+   * @param mixed $value
+   *   The value.
+   * @param bool $multiple
+   *   If set, multiple values will be returned as an array. Otherwise,
+   *   only the first value will be returned.
+   * @param string|null $image_style_name
+   *   An optional image style to apply if it exists.
+   *
+   * @return string[]|string|null
+   *   The image url(s).
+   */
+  public function getImage($value, $multiple = FALSE, $image_style_name = NULL) {
+    if (NULL === $value) {
+      return NULL;
+    }
+    if (isset($value[LANGUAGE_NONE])) {
+      $value = $value[LANGUAGE_NONE];
+    }
+    if (ParagraphHelper::isAssoc($value)) {
+      $value = [$value];
+    }
+    $uris = array_column($value, 'uri');
+    if (NULL !== $image_style_name && !empty(image_style_load($image_style_name))) {
+      $urls = array_map(function ($uri) use ($image_style_name) {
+        return image_style_url($image_style_name, $uri);
+      }, $uris);
+    }
+    else {
+      $urls = array_map([$this, 'getFileUrl'], $uris);
+    }
+
+    return $multiple ? $urls : reset($urls);
+  }
+
+  /**
+   * Get an absolute url from a "public:/" url.
+   *
+   * @param string $url
+   *   The file url.
+   *
+   * @return bool|string
+   *   The absolute url if any.
+   */
+  public function getFileUrl($url) {
+    return file_create_url($url);
+  }
+
+  /**
+   * Get ting identifiers.
+   *
+   * @param object $entity
+   *   The entity.
+   * @param string $field_name
+   *   The field name.
+   *
+   * @return string[]
+   *   A list of identifiers.
+   */
+  public function getTingIdentifiers($entity, $field_name) {
+    if (!isset($entity->{$field_name}[LANGUAGE_NONE])) {
+      return [];
+    }
+
+    $relations = $this->getFieldValue($entity, $field_name, 'value', TRUE);
+
+    $ids = array_map(function ($relation) {
+      if ((isset($relation->relation_type) && 'ting_reference' === $relation->relation_type)
+        && (isset($relation->endpoints[LANGUAGE_NONE][1]['entity_type']) && 'ting_object' === $relation->endpoints[LANGUAGE_NONE][1]['entity_type'])
+      ) {
+        return $relation->endpoints[LANGUAGE_NONE][1]['entity_id'];
+      }
+
+      return NULL;
+    }, $relations);
+
+    $result = db_query('SELECT o.ding_entity_id FROM {ting_object} o WHERE o.tid in (:ids)', [':ids' => $ids]);
+    $identifiers = $result->fetchCol();
+
+    return array_values($identifiers);
+  }
+
+  /**
+   * Get a ting identifier from a url.
+   *
+   * @param string $url
+   *   The url.
+   *
+   * @return string|null
+   *   The identifier if any.
+   */
+  public function getTingIdentifierFromUrl($url) {
+    return preg_match('@/object/(?P<identifier>.+)$@', $url, $matches) ? urldecode($matches['identifier']) : NULL;
```

Comment:
What if it is a collection url or is that not important here. Maybe the function should be renamed to `getPidFromTingObjectUrl`.